### PR TITLE
Add env values from Secret in the same way as EBS CSI driver does

### PR DESCRIPTION
**Pull request by stelucz**
### Issue

[Add env credential values from Secret #3191](https://github.com/kubernetes-sigs/aws-load-balancer-controller/issues/3191)

### Description

The EBS CSI Driver uses Secret (aws-secret) as env values source for AWS credentials. When CSI driver is deployed in a cluster it would be nice to reuse Secret in the same way for the LB controller. Ref:  https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/charts/aws-ebs-csi-driver/templates/controller.yaml#L104-L115

* Expanded env section in template to match usage in the EBS CSI driver.
* Added new helm value `envFromSecret` (default false) to instruct Helm to render env section based on Secret.
* Edited README to mention `envFromSecret` value.
* Manual tests (helm lint, helm test and deploy to live cluster).

### Checklist
- [ ] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `docs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Comment by M00nF1sh** [comment]
thanks for contributing. 
lgtm to me with one minor comment.

/ok-to-test

**Comment by johngmyers** [comment]
A better interface would be for the setting to specify the name of the Secret. Then the template should add the volume.

**Comment by stelucz** [comment]
and is lb controller able to use credential secret mounted as volume?

**Comment by johngmyers** [comment]
Ah, I misremembered how `secretKeyRef` worked. So no volume, but have the config option specify the name of the secret.

**Comment by stelucz** [comment]
@johngmyers secret name is configurable now

File: helm/aws-load-balancer-controller/README.md [hunk 1]
```diff
@@ -231,6 +231,7 @@ The default values set by the application itself can be confirmed [here](https:/
...
 | `livenessProbe`                                | Liveness probe settings for the controller                                                                                                                                                                             | (see `values.yaml`)                               |
 | `env`                                          | Environment variables to set for aws-load-balancer-controller pod                                                                                                                                                      | None                                              |
+| `envSecretName`                                | Environment variables credentials from Secret (aws-secret) for aws-load-balancer-controller pod similarly as The EBS CSI Driver does.                                                                               | `false`                                           |
```
Comment (johngmyers) [inline helm/aws-load-balancer-controller/README.md:234]: There is no default value. The description should not mention the EBS CSI Driver; just state that it takes the AWS credentials from the named secret. It should probably mention the need for `key_id` and `access_key` properties.

Comment (stelucz) [reply helm/aws-load-balancer-controller/README.md]: oh, I missed that. Thanks.

**Comment by stelucz** [comment]
Hi @johngmyers @M00nF1sh would be possible to merge it? Thanks

**Review by M00nF1sh — APPROVED** [review]
/lgtm
/approve
thanks for making this contribution!

**Pull request merged**